### ISSUE TITLE
Access logging of emitted dynamic metadata by ext_authz?

### ISSUE THREAD

**Issue by erikbos** [envoyproxy/envoy#12615]
I am trying to have an ext_authz authorization server emit dynamic metadata (https://github.com/envoyproxy/envoy/pull/11820) and log those fields in access log. 

I am using:
- envoy 1.15.0
- go-control-plane github.com/envoyproxy/go-control-plane v0.9.7-0.20200811182123-112a4904c4b0 with the following test code snippet:

```
     metadata := &structpb.Struct{
		Fields: map[string]*structpb.Value{
			"Customer": {
				Kind: &structpb.Value_StringValue{
					StringValue: "VIP",
				},
			},
		},
	}
	response := &authservice.CheckResponse{
		Status: &status.Status{
			Code:    int32(rpc.OK),
			Message: "Hello World",
		},
		HttpResponse: &authservice.CheckResponse_OkResponse{
			OkResponse: &authservice.OkHttpResponse{
				// Headers:         buildHeadersList(headers),
				DynamicMetadata: metadata,
			},
		},
	}
	log.Printf("OkHttpResponse: %+v", response)
```

filter chain:
```
          "http_filters": [
             {
              "name": "envoy.filters.http.ext_authz",
              "typed_config": {
               "@type": "type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz",
               "grpc_service": {
                "envoy_grpc": {
                 "cluster_name": "authentication"
                },
                "timeout": "2.120s"
               },
               "transport_api_version": "V3"
              }
             },
             {
              "name": "envoy.filters.http.cors"
             },
             {
              "name": "envoy.filters.http.router"
             }
```

With envoyproxy accesslog configured:

```
"access_log": [
             {
              "name": "envoy.access_loggers.file",
              "typed_config": {
               "@type": "type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog",
               "path": "/dev/stdout",
               "log_format": {
                "json_format": {
                 "request_path": "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%",
                 "response_flags": "%RESPONSE_FLAGS%",
                 "upstream_response_time": "%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%",
                 "downstream_remote_address": "%DOWNSTREAM_REMOTE_ADDRESS%",
                 "authz": "%DYNAMIC_METADATA(envoy.filters.http.ext_authz)%",
                 "response_code": "%RESPONSE_CODE%",
                 "request_id": "%REQ(REQUEST-ID)%",
                 "start_time": "%START_TIME%",
                 "request_method": "%REQ(:METHOD)%",
                 "upstream_host": "%UPSTREAM_HOST%",
                 "caller": "%REQ(CALLER)%",
                 "http2_authority": "%REQ(:AUTHORITY)%",
                 "bytes_received": "%BYTES_RECEIVED%",
                 "content_type": "%REQ(CONTENT-TYPE)%",
                 "method": "%REQ(:METHOD)%",
                 "x_forwarded_for": "%REQ(X-FORWARDED-FOR)%",
                 "route_name": "%ROUTE_NAME%",
                 "customer": "%DYNAMIC_METADATA(envoy.filters.http.ext_authz:Customer)%",
                 "bytes_sent": "%BYTES_SENT%",
                 "client_address": "%DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT%",
                 "request_duration": "%DURATION%",
                 "upstream_cluster": "%UPSTREAM_CLUSTER%",
                 "user_agent": "%REQ(USER-AGENT)%",
                 "response_duration": "%RESPONSE_DURATION%",
                 "protocol": "%PROTOCOL%",
                 "upstream_transport_failure_reason": "%UPSTREAM_TRANSPORT_FAILURE_REASON%",
                 "path": "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"
                }
               }
              }
             }
            ],
```
field `authz` & `customer` has been set to log the right metadata namespace & key I think.

Envoyproxy logs:

```
[2020-08-12 19:40:47.952][4809889][debug][router] [external/envoy/source/common/router/router.cc:583] [C0][S4266729642886229251] router decoding headers:
':method', 'POST'
':path', '/envoy.service.auth.v3.Authorization/Check'
':authority', 'authentication'
':scheme', 'http'
'te', 'trailers'
'grpc-timeout', '2120m'
'content-type', 'application/grpc'
'x-envoy-internal', 'true'
'x-forwarded-for', '192.168.178.34'
'x-envoy-expected-rq-timeout-ms', '2120'

[2020-08-12 19:40:47.952][4809889][debug][pool] [external/envoy/source/common/http/conn_pool_base.cc:71] queueing request due to no available connections
[2020-08-12 19:40:47.952][4809889][debug][pool] [external/envoy/source/common/conn_pool/conn_pool_base.cc:53] creating a new connection
[2020-08-12 19:40:47.952][4809889][debug][client] [external/envoy/source/common/http/codec_client.cc:35] [C9] connecting
[2020-08-12 19:40:47.952][4809889][debug][connection] [external/envoy/source/common/network/connection_impl.cc:753] [C9] connecting to 127.0.0.1:4000
[2020-08-12 19:40:47.952][4809889][debug][connection] [external/envoy/source/common/network/connection_impl.cc:769] [C9] connection in progress
[2020-08-12 19:40:47.952][4809889][debug][http2] [external/envoy/source/common/http/http2/codec_impl.cc:1063] [C9] updating connection-level initial window size to 268435456
[2020-08-12 19:40:47.952][4809889][debug][connection] [external/envoy/source/common/network/connection_impl.cc:616] [C9] connected
[2020-08-12 19:40:47.952][4809889][debug][client] [external/envoy/source/common/http/codec_client.cc:73] [C9] connected
[2020-08-12 19:40:47.952][4809889][debug][pool] [external/envoy/source/common/conn_pool/conn_pool_base.cc:146] [C9] attaching to next request
[2020-08-12 19:40:47.952][4809889][debug][pool] [external/envoy/source/common/conn_pool/conn_pool_base.cc:73] [C9] creating stream
[2020-08-12 19:40:47.952][4809889][debug][router] [external/envoy/source/common/router/upstream_request.cc:342] [C0][S4266729642886229251] pool ready
[2020-08-12 19:40:47.954][4809889][debug][router] [external/envoy/source/common/router/router.cc:1153] [C0][S4266729642886229251] upstream headers complete: end_stream=false
[2020-08-12 19:40:47.954][4809889][debug][http] [external/envoy/source/common/http/async_client_impl.cc:99] async http request response headers (end_stream=false):
':status', '200'
'content-type', 'application/grpc'
'x-envoy-upstream-service-time', '1'

[2020-08-12 19:40:47.954][4809889][debug][client] [external/envoy/source/common/http/codec_client.cc:105] [C9] response complete
[2020-08-12 19:40:47.954][4809889][debug][pool] [external/envoy/source/common/conn_pool/conn_pool_base.cc:98] [C9] destroying stream: 0 remaining
[2020-08-12 19:40:47.954][4809889][debug][http] [external/envoy/source/common/http/async_client_impl.cc:125] async http request response trailers:
'grpc-status', '0'
'grpc-message', ''

[2020-08-12 19:40:47.954][4809889][debug][router] [external/envoy/source/common/router/router.cc:426] [C8][S3843439789292979697] cluster 'people_localhost' match for URL `x`
```

Accesslog fields stays empty:
```
"authz":null,"customer":null
```

I had hoped/expected the returned metadata by authz to appear in envoy's debug logging and/or access log..

Could someone point me in the right direction to get this to work?

**Comment by erikbos** [comment]
Oh crap, I just notice #11820 was merged before `envoy 1.15.0` was released, let me master :)

**Comment by erikbos** [comment]
The above configuration works when running a version of Envoy containing #11820 👍
